src/gdpr/index.ts

```diff
@@ -1,28 +1,73 @@
 import { TCData } from '../../types';
 
-const hasConsent = (gdprApplies?: boolean): boolean => {
-  return gdprApplies === undefined ? false : gdprApplies;
+const hasGdprConsent = (vendorIds: number[], tcData: TCData): boolean => {
+  const { gdprApplies, vendor } = tcData;
+  console.log('HAS GDPR CONSENT', { gdprApplies, vendor });
+  return (
+    gdprApplies === false ||
+    (vendorIds.length > 0 &&
+      vendorIds.every((vendorId) => !!vendor.consents[vendorId]))
+  );
 };
 
-export const hasGdprConsent = (vendorIds?: number[]): Promise<boolean> => {
+const removeListener = (tcData: TCData): Promise<boolean> => {
   return new Promise((resolve) => {
-    // If the api is missing then assume no consent
-    if (window.__tcfapi === undefined) resolve(false);
-    else {
+    if (tcData.listenerId) {
       window.__tcfapi(
-        'getTCData',
+        'removeEventListener',
         2,
-        ({ gdprApplies, vendor }: TCData, success: boolean) => {
-          if (success && vendorIds) {
-            resolve(
-              hasConsent(gdprApplies) &&
-                vendorIds.every((vendorId) => !!vendor.consents[vendorId])
-            );
-          } else {
-            resolve(false);
-          }
-        }
+        (success) => resolve(success),
+        tcData.listenerId
       );
     }
   });
 };
+
+// This promise will only resolve once there is gdpr consent at that point in time
+export const checkForConsent = (
```

Comment:
I think its confusing to have `checkForConsent` take an arg of `waitForConsent` and then for this to be called by `waitOnConsent`.

Why not just have one function which is `waitForConsent`, is there a use case in which we do not need to wait for consent? I do not think there is or should be - if its for tests I think we need to try a different approach there.